src/pencilkit.cs

```diff
@@ -246,9 +300,171 @@ interface PKToolPicker {
 		PKToolPicker GetSharedToolPicker (UIWindow window);
+
+		[iOS (14, 0)]
+		[Introduced (PlatformName.MacCatalyst, 14, 0)]
+		[Export ("showsDrawingPolicyControls")]
+		bool ShowsDrawingPolicyControls { get; set; }
+
+		// !missing-null-allowed! 'System.Void PencilKit.PKToolPicker::set_AutosaveName(System.String)' is missing an [NullAllowed] on parameter #0
+		[iOS (14, 0)]
+		[Introduced (PlatformName.MacCatalyst, 14, 0)]
+		[Export ("stateAutosaveName")]
+		string stateAutosaveName { get; set; }
+	}
+
+	[Mac (10, 16), iOS (14, 0)]
+	[Introduced (PlatformName.MacCatalyst, 14, 0)]
+	[BaseType (typeof(NSObject))]
+	interface PKInk : NSCopying
+	{
+		[Export ("initWithInkType:color:")]
+		[DesignatedInitializer]
+		IntPtr Constructor (string type, UIColor color);
+
+		[Export ("inkType")]
+		string InkType { get; }
+
+		[Export ("color")]
+		UIColor Color { get; }
+	}
+
+	[Mac (10, 16), iOS (14, 0)]
+	[Introduced (PlatformName.MacCatalyst, 14, 0)]
+	[BaseType (typeof(NSObject))]
+	interface PKFloatRange : NSCopying
+	{
+		[Export ("initWithLowerBound:upperBound:")]
+		IntPtr Constructor (nfloat lowerBound, nfloat upperBound);
+
+		[Export ("lowerBound")]
+		nfloat LowerBound { get; }
+
+		[Export ("upperBound")]
+		nfloat UpperBound { get; }
+	}
+
+	[Mac (10, 16), iOS (14, 0)]
+	[Introduced (PlatformName.MacCatalyst, 14, 0)]
+	[BaseType (typeof(NSObject))]
+	interface PKStroke : NSCopying
+	{
+		[NoMac]
+		[Export ("initWithInk:strokePath:transform:mask:")]
+		IntPtr Constructor (PKInk ink, PKStrokePath path, CGAffineTransform transform, [NullAllowed] UIBezierPath mask);
+
+		[NoiOS]
+		[Export ("initWithInk:strokePath:transform:mask:")]
+		IntPtr Constructor (PKInk ink, PKStrokePath path, CGAffineTransform transform, [NullAllowed] NSBezierPath mask);
+
+		[Export ("ink")]
+		PKInk Ink { get; }
+
+		[Export ("transform")]
+		CGAffineTransform Transform { get; }
+
+		[Export ("path")]
+		PKStrokePath Path { get; }
+
+#if MONOMAC
+		[NoiOS]
+		[NullAllowed, Export ("mask")]
+		NSBezierPath Mask { get; }
+#else
+		[NoMac]
+		[NullAllowed, Export ("mask")]
+		UIBezierPath Mask { get; }
+#endif
+
+		[Export ("renderBounds")]
+		CGRect RenderBounds { get; }
+
+		[Export ("maskedPathRanges")]
+		PKFloatRange[] MaskedPath { get; }
+	}
+
+	delegate void PKInterpolatedPointsEnumeratorHandler (PKStrokePoint strokePoint, out bool stop);
+
+	[Mac (10, 16), iOS (14, 0)]
+	[Introduced (PlatformName.MacCatalyst, 14, 0)]
+	[BaseType (typeof(NSObject))]
+	interface PKStrokePath : NSCopying
+	{
+		[Export ("initWithControlPoints:creationDate:")]
+		[DesignatedInitializer]
+		IntPtr Constructor (PKStrokePoint[] controlPoints, NSDate creationDate);
+
+		[Export ("count")]
+		nuint Count { get; }
+
+		[Export ("creationDate")]
+		NSDate CreationDate { get; }
+
+		[Export ("pointAtIndex:")]
+		PKStrokePoint PointAtIndex (nuint i);
+
+		[Export ("objectAtIndexedSubscript:")]
+		PKStrokePoint GetObjectAtIndexedSubscript (nuint i);
```

Comment:
```suggestion
		PKStrokePoint GetObject (nuint indexedSubscript);
```

Reply:
This object is also a good candidate for a read-only [C# indexer](https://docs.microsoft.com/en-us/dotnet/csharp/programming-guide/indexers/) using `GetObject`, this is additional sugar is not really needed but a nice to have.